groovy-mode.el

```diff
@@ -329,7 +312,77 @@ The function name is the second group in the regexp.")
   (defconst groovy-dollar-slashy-open-regex
     (rx "$/"))
   (defconst groovy-dollar-slashy-close-regex
-    (rx "/$")))
+    (rx "/$"))
+  (defconst groovy-declaration-regex
+    (rx-to-string
+     `(seq
+       line-start
+       (* space)
+       (+
+        (or "def" "public" "private" "protected" "final" "static"
+            (seq "@" (+ alphanumeric))
+            (regexp ,groovy-type-regexp))
+        (not (any "."))))))
+  (defconst groovy-variable-regex
+    (rx-to-string
+     `(seq
+       (seq (group (or bol space "(" ","))
+            (regexp ,groovy-symbol-regexp))
+       (* space)
+       (group (or "=" ";" ")" "," "->" "//" "/*" eol))))))
+
+(defun groovy--variable-names-search (limit)
+  "Search for variable names up to LIMIT."
+  (let* ((pos (point))
+         (matched (re-search-forward groovy-variable-regex limit t)))
+    (when (and matched (> matched pos))
+      (when (or (equal (match-string 3) "//")
+                (equal (match-string 3) "/*"))
+        (setq matched (- matched 2)))
+      (or (and (not (groovy--in-string-p))
+               (not (groovy--comment-p matched))
+               (let* ((line-beg (- (match-beginning 0) (line-beginning-position)))
+                      (cur-line (groovy--current-line))
+                      (str (buffer-substring-no-properties (line-beginning-position) (match-beginning 0)))
+                      (match-str (s-trim (match-string 0)))
+                      (open-char (match-string 1)))
+                 (message "match-str: %s" match-str)
+                 (save-match-data
+                   (unless (string-match (rx bol (* space) (eval match-str) (* space) eol) cur-line)
```

Comment:
This isn't quite right: `rx` is evaluated at compile time, and match-str isn't defined at that point. I think you want to use `rx-to-string` with backquoting.